cmake/find/FindJulia.cmake

```diff
@@ -0,0 +1,131 @@
+#
+# This FindJulia.cmake is intended to detect julia as a package.
+#
+# Use
+# `list(APPEND CMAKE_MODULE_PATH "path/to/this/file")`
+# `find_package(Julia 1.7.0 REQUIRED)`
+#
+# to make the julia target available through:
+# `"$<BUILD_INTERFACE:Julia::Julia>"`
```

Comment:
The use of the generator expression was explained in the original PR.

Also, if you wish to comment the find module, then you should use the CMake convention of inline reStructuredText comment:  
https://github.com/Kitware/CMake/blob/master/Modules/FindThreads.cmake  
https://github.com/friendlyanon/cmake-init-use-pkg-config/blob/master/cmake/find/Findgumbo.cmake